docs/iam-usage.rst

```diff
@@ -0,0 +1,144 @@
+Using IAM For Users
+===================
+
+.. note:: This document does not explain how Google Cloud IAM works,
+   for that, please check out `the docs <https://cloud.google.com/iam/docs/>`_.
+
+Resources that implement the IAM interface provide the following methods:
+
+Policy Editing Methods
+----------------------
+
+.. note:: All policy editing methods except ``set_policy`` have an optional
+   ``retries`` keyword argument that takes the number of times to retry in the
+   even of an interrupted transaction
+
+>>> resource.add_roles(iam.user('alice@example.com'), iam.roles.OWNER.name, iam.roles.EDITOR.name)
+True
```

Comment:
Also to explain. IAM is a meta-api, so it's a common interface implemented by many APIs (e.g. storage, pub/sub, GCE, CRM). so here: `resource` refers to an instance of a type that inherits from the mixin, (and is backed by a One Platform "Resource")